src/libraries/System.Private.CoreLib/src/System/Number.NumberBuffer.cs

```diff
@@ -29,6 +29,25 @@ internal static partial class Number
             public NumberBufferKind Kind;
             public Span<byte> Digits;
 
+            public NumberBuffer(NumberBufferKind kind, Span<byte> digits)
+            {
+                Debug.Assert(digits != null);//TODO we do still need it?
```

Comment:
Span is a struct, so this assert cannot ever fire.

Reply:
tannergooding told me the same: https://github.com/dotnet/runtime/pull/38696#discussion_r449070475
But I'm afraid it can compare to null, because null implicitly casts to array, so this code return true:

            static bool X(ref ReadOnlySpan<char> str)
            {
                if (str == null)
                {
                    return true;
                }
                return false;
            }

            ReadOnlySpan<char> str = /*(char[])*/null;
            Console.WriteLine(X(ref str));
            return;